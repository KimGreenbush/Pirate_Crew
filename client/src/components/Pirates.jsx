import React from "react"
import DeleteButton from "../components/DeleteButton"
import {Link} from "@reach/router"

const Pirates = ({ pirates, setPirates}) => {
    const removePirate = pirateId => {
        setPirates(pirates.filter(pirate => pirate._id !== pirateId))
    }

    const style = {
        width: "100px"
    }

    return (
        <div>
            {pirates.sort((a, b) => a.name.localeCompare(b.name)).map((pirate, idx) =>
                <div key={idx}>
                    <img src={pirate.image} style={style} alt="pirate"/>
                    <p>{pirate.name}</p>
                    <p><Link to={`/pirate/${pirate._id}`}><button>View Pirate</button></Link> <DeleteButton pirateId={pirate._id} successCallback={() => removePirate(pirate._id)} /></p>
                </div>)}
        </div>
    )
}

export default Pirates