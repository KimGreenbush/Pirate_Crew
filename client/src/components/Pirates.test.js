/**
 * @jest-environment jsdom
 */

import React from "react";
import Pirates from "./Pirates";
import renderer from "react-test-renderer";

test("renders Pirates without crashing", () => {
	const component = renderer.create(<Pirates />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});