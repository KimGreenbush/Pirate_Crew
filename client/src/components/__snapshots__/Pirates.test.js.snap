// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pirates without crashing 1`] = `
<div
  className="pirates"
>
  
</div>
`;
